### PR TITLE
Prototype for Image Widget settings [stage-1]

### PR DESCRIPTION
**Pull request by alex-deaconu**

### CODE REVIEW

File: web/scripts/widgets/services/svc-widget-settings-factory.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          }
+        }
+      };
+
+      factory.showSettingsModal = function (item, softUpdate) {
+        if (!item || !item.objectReference && !item.settingsUrl) {
+          return;
+        }
+
+        var modalInstance = $modal.open({
```
Comment (ezequielc) [inline web/scripts/widgets/services/svc-widget-settings-factory.js]: @alex-deaconu How do we show the templates for other Widgets? I assume they all use the same controller widgetSettingsModal, but the templateUrl needs to be changed, right? 
Should we have another if statement here? Thanks

Comment (alex-deaconu) [reply web/scripts/widgets/services/svc-widget-settings-factory.js]: @ezequielc the templateUrl String will be part of the configuration for the Widget. Thanks.

Comment (ezequielc) [reply web/scripts/widgets/services/svc-widget-settings-factory.js]: @alex-deaconu Sounds good. Thanks!

**Comment by alex-deaconu** [comment]
This PR does should actually change anything in Apps. I've disabled the In App UI for the Image Widget, so now it just creates the infrastructure required for the Twitter Widget (separate branch/PR).

The risevision.widgets.* files are imported from https://github.com/Rise-Vision/widget-settings-ui-core & https://github.com/Rise-Vision/widget-image. I did a bit of refactoring, but mostly added unit test coverage.

@ezequielc @fjvallarino @rudibravo please review. @rodrigopavezi FYI. Thanks.

**Comment by fjvallarino** [comment]
@alex-deaconu the code looks good, but I wonder if there's any way I can test it in order to better understand what changed/was added.

**Comment by alex-deaconu** [comment]
@fjvallarino unfortunately the only Widget that was working with this prototype is the Image Widget, but that was disabled. 

I think the most important thing to test at this point is that the existing (iframe) widget and embedded presentations still work correctly (getting and saving the settings). I'm going to try to re-stage the previous commit before disabling the Image Widget in App settings. Will link the URL. Thanks.

**Comment by alex-deaconu** [comment]
@fjvallarino I've restaged the previous commit at https://apps-stage-1.risevision.com. Now the Image Widget settings will open inApp. Thanks.

**Comment by fjvallarino** [comment]
@alex-deaconu looks good

**Pull request merged**